File: src/main/java/juuxel/woodsandmires/feature/LakeFeatureExtensions.java
package juuxel.woodsandmires.feature;

import net.minecraft.block.BlockState;
import org.jetbrains.annotations.Nullable;

import java.util.Random;

public interface LakeFeatureExtensions {
    boolean wam_shouldReplaceBorderingBlocks();
    @Nullable BlockState wam_getBorderBlock(Random random);
}

File: src/main/kotlin/juuxel/woodsandmires/feature/LakeFeatureExtensions.kt
package juuxel.woodsandmires.feature

import net.minecraft.block.BlockState
import java.util.Random

@Suppress("FunctionName")
interface LakeFeatureExtensions {
    fun wam_shouldReplaceBorderingBlocks(): Boolean
    fun wam_getBorderBlock(random: Random): BlockState?
}
